src/views/login/components/wallet-button-list/wallet-button-list.view.tsx

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+
+import useWalletButtonListStyles from "src/views/login/components/wallet-button-list/wallet-button-list.styles";
+import WalletButton from "src/views/login/components/wallet-button/wallet-button.view";
+import { WalletName } from "src/store/login/login.actions";
+
+declare const window: Window & typeof globalThis & { ethereum?: unknown };
```

Comment:
Can we move this to the `react-app-env.d.ts` file?

Reply:
I'm not sure we can export/import a `const` declaration. Will try to.

Reply:
Wait, in `react-app-env.d.ts`? But this is not a type, it's the declaration of a `const`.

Or are you referring to creating the type and then declaring the const on each module?

I'm a bit lost.

Reply:
Thx for the `BaseProvider` type for the ethereum prop! I spent some time looking for it and did not succeed.

I've committed the closer I can get using a shared global type + local declarations:

```diff
diff --git a/src/react-app-env.d.ts b/src/react-app-env.d.ts
index 6431bc5..01a8fa5 100644
--- a/src/react-app-env.d.ts
+++ b/src/react-app-env.d.ts
@@ -1 +1,9 @@
 /// <reference types="react-scripts" />
+
+import { BaseProvider } from "@ethersproject/providers";
+declare global {
+  type WindowOverride = Window &
+    typeof globalThis & {
+      ethereum?: BaseProvider;
+    };
+}
diff --git a/src/views/app.view.tsx b/src/views/app.view.tsx
index 12f0f35..ab21743 100644
--- a/src/views/app.view.tsx
+++ b/src/views/app.view.tsx
@@ -25,12 +25,7 @@ import {
 } from "src/domain/hermez";
 import { EthereumNetwork } from "src/domain/ethereum";

-declare const window: Window &
-  typeof globalThis & {
-    ethereum?: {
-      on: (event: "accountsChanged" | "chainChanged", callback: () => void) => void;
-    };
-  };
+declare const window: WindowOverride;

 interface AppStateProps {
   wallet: HermezWallet.HermezWallet | undefined;
diff --git a/src/views/login/components/wallet-button-list/wallet-button-list.view.tsx b/src/views/login/components/wallet-button-list/wallet-button-list.view.tsx
index 1ddd7b1..9ee15d2 100644
--- a/src/views/login/components/wallet-button-list/wallet-button-list.view.tsx
+++ b/src/views/login/components/wallet-button-list/wallet-button-list.view.tsx
@@ -4,7 +4,7 @@ import useWalletButtonListStyles from "src/views/login/components/wallet-button-
 import WalletButton from "src/views/login/components/wallet-button/wallet-button.view";
 import { WalletName } from "src/store/login/login.actions";

-declare const window: Window & typeof globalThis & { ethereum?: unknown };
+declare const window: WindowOverride;

 interface WalletButtonListProps {
   onClick: (walletName: WalletName) => void;
```